notebook/static/notebook/js/menubar.js

```diff
@@ -87,6 +88,7 @@ define([
         }
     };
 
+    // deprecate ?
```

Comment:
I'm unaware of anyone who'd subclass instead of listening to the event.  Regardless, it's probably okay to deprecate since the event basically allows the same extensibility.